src/Flysystem/FlysystemAssetStore.php

```diff
@@ -300,6 +301,9 @@ private function applyToFileOnFilesystem(callable $callable, ParsedFileID $parse
             self::VISIBILITY_PROTECTED
         ];
 
+        /** @var FileHashingService $hasher */
+        $hasher = Injector::inst()->get(FileHashingService::class);
```

Comment:
But I like new APIs!!!

I've made a conscious decision to leave caching disabled by default because I didn't want to think of all the way this could go wrong if it was on all the time.

However, I think we should seriously consider turning caching on all the time in SS4.5 once we have more time to think through all the implication. Regular file requests will have the same problem that the FileMigrationTask has. It might not be that apparent because you're only working with one file.

Reply:
I think once we addressed the regression from 4.3 about computing hashes on non-write operations in FlysystemAssetStore (see separate comment), then I don't think we should turn on this caching by default. It's only important for batch write operations on *file contents*, right? Even if you upload a dozen files at once, the overhead on that is relatively small. Caching adds complexity, and we mainly need it for file migration tasks.

Reply:
Nothing else to do here.